api/test/wfl/integration/datarepo_test.clj

```diff
@@ -3,59 +3,195 @@
+   :replicate_concordant_sites                   29857
+   :align_to_self_merged_reads_aligned           315194
+   :dist_to_ref_indels                           0
+   :replicate_discordant_snps                    0
+   :ivar_version                                 "iVar version 1.3"
+   :assembly_fasta                               "gs://broad-gotc-dev-wfl-ptc-test-inputs/covid-19/assemble_refbased/call-call_consensus/MA_MGH_02760.lERCC-00134_RandomPrimer-SSIV_NexteraXT_RNA_0352325199_cDNA_0369438842_LIB_13215203_G2_HTHC3DRXX_L1.cleaned.fasta"
+   :align_to_self_merged_mean_coverage           1064.595324883791
+   :align_to_ref_merged_reads_aligned            315074
+   :align_to_ref_per_input_fastqc                ["gs://broad-gotc-dev-wfl-ptc-test-inputs/covid-19/assemble_refbased/call-align_to_ref/shard-0/MA_MGH_02760.lERCC-00134_RandomPrimer-SSIV_NexteraXT_RNA_0352325199_cDNA_0369438842_LIB_13215203_G2_HTHC3DRXX_L1.cleaned.mapped_fastqc.html"
+                                                  "gs://broad-gotc-dev-wfl-ptc-test-inputs/covid-19/assemble_refbased/call-align_to_ref/shard-1/MA_MGH_02760.lERCC-00134_RandomPrimer-SSIV_NexteraXT_RNA_0352325199_cDNA_0369438842_LIB_13215203_G2_HTHC3DRXX_L2.cleaned.mapped_fastqc.html"]})
+
+(def ^:private genomics-mime-types
+  {"bam"   "application/octet-stream"
+   "cram"  "application/octet-stream"
+   "fasta" "application/octet-stream"
+   "vcf"   "text/plain"})
+
+(defn ^:private run-thunk! [x] (x))
+
+(defn ^:private ingest!
+  "Ingest `value` into the dataset specified by `dataset-id` depending on
+   the `value`'s WDL `type` and return a thunk that performs any delayed work
+   and returns an ingest-able value."
+  [workload-id dataset-id profile-id type value]
+  ;; Ingesting objects into TDR is asynchronous and must be done in two steps:
+  ;; - Issue an "ingest" request for that object to TDR
+  ;; - Poll the result of the request for the subsequent resource identifier
+  ;;
+  ;; Assuming TDR can fulfil ingest requests in parallel, we can (in theory)
+  ;; increase throughput by issuing all ingest requests up front and then
+  ;; poll for the resource identifiers later.
+  ;;
+  ;; To do this, this function returns a thunk that when run, performs
+  ;; any delayed work needed to ingest an object of that data type (such as
+  ;; polling for a file resource identifier) and returns a value that can
+  ;; be ingested into a dataset table.
+  (let [sequence    (fn [xs] #(mapv run-thunk! xs))
+        return      (fn [x]   (constantly x))
+        bind        (fn [f g] (comp g f))
+        ingest-file (partial datarepo/ingest-file dataset-id profile-id)]
+    ((fn go [type value]
+       (case (:typeName type)
+         "File"
+         (let [[bkt obj] (gcs/parse-gs-url value)
+               target    (str/join "/" ["" workload-id obj])
+               mime      (mime-type/ext-mime-type value genomics-mime-types)]
+           (-> (get-in env/stuff [:debug :data-repo :service-account])
```

Comment:
Just fyi, I took the `stuff` and `:debug` out from the `environment.clj` so this needs to be something like `(env/getenv "WFL_DATA_REPO_SA")` now

Reply:
Thanks! Merged with your changes now!